### PR TITLE
Fix typo in iterated expectation equation

### PR DESCRIPTION
**Pull request by zaxtax**
#### Submission Checklist

- [x] Builds locally
- [x] New functions marked with `` <<{ since VERSION }>>``
- [x] Declare copyright holder and open-source license: see below

#### Summary
Fixes a typo

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): I am the copyright holder

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC BY-ND 4.0 (https://creativecommons.org/licenses/by-nd/4.0/)

### CODE REVIEW

File: src/stan-users-guide/latent-discrete.qmd [hunk 1]
```diff
@@ -1063,7 +1063,7 @@ $$
 $$
 This estimator is justified by the law of iterated expectation, the fact that
 $$
-\mathbb{E}[h(\Theta)] = \mathbb{E}[\mathbb{E}[g(\Theta, Z] \mid \Theta)] = \mathbb{E}[g(\Theta, Z)] = \mu.
+\mathbb{E}[h(\Theta)] = \mathbb{E}[\mathbb{E}[g(\Theta, Z)] \mid \Theta] = \mathbb{E}[g(\Theta, Z)] = \mu.
```
Comment (bob-carpenter) [inline src/stan-users-guide/latent-discrete.qmd:1066]: There's still a stray close square bracket on 1066 after `Z)`.

**Review by bob-carpenter — CHANGES_REQUESTED** [review]
Thanks, Rob!  The problem is that the `Z[` should have been `Z)`.

File: src/stan-users-guide/latent-discrete.qmd (hunk 1 above)
Comment (zaxtax) [reply src/stan-users-guide/latent-discrete.qmd:1066]: Are you sure? it's that close square bracket close the inner Expectation?

Comment (bob-carpenter) [reply src/stan-users-guide/latent-discrete.qmd:1066]: Oops.  You're right.  Thanks for your patience. As you can see, I have a bit of an issue with paren balancing---I usually just let emacs tell me :-).

Comment (zaxtax) [reply src/stan-users-guide/latent-discrete.qmd:1066]: I cut-paste it into emacs with rainbow delimiters just to triple-check 😅

**Pull request merged**